### PR TITLE
fix(node-resolve): handle browser-mapped paths correctly in nested contexts

### PR DESCRIPTION
**Pull request by eemeli**
## Rollup Plugin Name: `node-resolve`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers: #919, eemeli/yaml#291

### Description

For each entry in a package.json `"browser"` object entry, the `browserMapCache` is filled with mapped paths at keys using both the raw source of the path that needs to be remapped, as well as the resolved path. This effectively enables both package name imports (like `"foo"`) as well as relative path imports (like `"./foo"`) to work when `doResolveId` is called. However, this causes a problem when the raw form of a relative import in a sub-path matches a value that's included in the package.json `"browser"` entry.

In other words, if `"browser"` includes `"./foo": "./other.js"`, an import of `"./foo"` in a file `./bar/baz.js` is mapped to `./other.js`, even though it shouldn't; [the spec](https://github.com/defunctzombie/package-browser-field-spec) is pretty clear on this:

> All paths for browser fields are relative to the `package.json` file location (and usually project root as a result).

To fix, only the resolved paths are checked when applying mappings for relative imports.

### CODE REVIEW

**Comment by guybedford** [comment]
Thanks you for this fix and apologies for the delayed review! This looks great to me.

**Review by guybedford — APPROVED** [review]
